pubsub.go

```diff
@@ -72,6 +75,18 @@ type PubSub struct {
 	// a notification channel for when our peers die
 	peerDead chan peer.ID
 
+	// tracks the channels subscribed to new peer notifications
+	newPeerNotif map[chan peer.ID]struct{}
+
+	// setup new peer notifier
+	addPeerNotif chan *reqNewPeerNotifier
+
+	// remove peer notifier
+	rmPeerNotif chan chan peer.ID
```

Comment:
and this?

Reply:
as with any event subscription service you should be able to unsubscribe. this is called when we're done bootstrapping saying "no need to tell me about peer connections anymore"

Reply:
this is unnecessary.